### PR TITLE
Split the various modules of the build script

### PR DESCRIPTION
**Pull request by guruDanny67**
This should permit a customization / grouping of the projects/tools to build

Second try, even in this case I put the note on the NotIplemented exception for the project build and the check for the duplicate project.

### CODE REVIEW

**Comment by nacho** [comment]
looks pretty good to me now

**Comment by guruDanny67** [comment]
I think it's a good compromise between splitting 'everithing' (long time ago, when we use the original RCS system, the rule was 'one module (C language) one function!' to avoid lock problems) and have only one module with all in it.

In the end to expand the build system you normally should add some classes to projects  or tools module (or make a new one) and don't care about the low level stuff into utils, where we can add base functionality to help us do the job (I was thinking about making a class for the message/log/debug system, with a couple of logging for what's build and the time spent do make the build, for examples)

To summarize all, I think is ok to merge this PR and grow from this structure, but my opionion is a little bit biased :)

**Comment by nacho** [comment]
Thanks for working on this
